### ISSUE TITLE
Linux Boost version mismatch for CXX20

### ISSUE THREAD

**Issue by raneamri** [auto-differentiation/QuantLib-Risks-Cpp#22]
QuantLib expects Boost ver. ≥1.75.0 for CXX ≥20 but Linux fetches 1.74.0 with `sudo apt install -y libboost-dev ccache ninja-build`:

```
CMake Error at /usr/local/share/cmake-3.30/Modules/FindPackageHandleStandardArgs.cmake:233 (message):
  Could NOT find Boost: Found unsuitable version "1.74.0", but required is at
  least "1.75.0" (found /usr/include, )
```

**Comment by xcelerit-team** [comment]
That depends on the Linux distribution and version and how it's setup I guess... Which Linux version are you using?

**Comment by xcelerit-team** [comment]
Their development images used for building quantlib [install boost from scratch](https://github.com/lballabio/QuantLib/blob/master/dockerfiles/ci.Dockerfile#L15). And the version used currently in their Linux builds is [`rolling`](https://github.com/lballabio/QuantLib/blob/master/.github/workflows/linux.yml#L57). (see also [here](https://github.com/lballabio/QuantLib/blob/master/.github/workflows/linux.yml#L136))

The container images can be found [here](https://github.com/lballabio/QuantLib/pkgs/container/quantlib-devenv/versions?filters%5Bversion_type%5D=tagged), and we see it's using Boost 1.85.

Perhaps we should try to use the same container for our CI builds that QuantLib is using - that would make things easier to stay compatible....

**Comment by xcelerit-team** [comment]
(and note that you can set the standard using the CMake command-line, i.e. add `-DCMAKE_CXX_STANDARD=20` to the cmake configure call)

**Comment by raneamri** [comment]
Good idea, I'll match the Linux workflow to QuantLib's. Is it also worth adding matrix configuration for CXX 20?

**Comment by raneamri** [comment]
Should be fixed by #23
